### PR TITLE
RED-119 add github ci

### PR DESCRIPTION
**Pull request by TieDyedSheep**
DO NOT MERGE without approval from Andrew!

### CODE REVIEW

**Review by paulgs9988 — DISMISSED** [review]
Similar issue to the review I left on the Shardeum ci.yml PR:

We have "build:dev" but not "build" on the shardus-core repo so we get a failure from the github automation "QA merge checks", but unrelated to this guy here.

File: package.json [hunk 1]
```diff
@@ -34,6 +33,7 @@
     "release:move": "npm-run-all release:move:*",
     "build:dev": "npm-run-all clean compile:ts dev:move",
     "build:release": "npm-run-all clean prep compile release:move",
+    "build": "npm run compile",
```
Comment (chrypnotoad) [inline package.json]: don't need this anymore

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    uses: shardeum/github-automation/.github/workflows/reusable-node-ci.yml@dev
+    permissions:
+      issues: write
+      pull-requests: write
+      contents: write
+    with:
+      node-version: ${{ vars.NODE_VERSION }}
+      lint-required: ${{ vars.IS_LINT_REQUIRED == 'true' }}
+      format-check-required: ${{ vars.IS_FORMAT_CHECK_REQUIRED == 'true' }}
+      apply-patches-required: ${{ vars.IS_APPLY_PATCHES_REQUIRED == 'true' }}
```
Comment (chrypnotoad) [inline .github/workflows/ci.yml:60]: should be false

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    if: ${{ github.event.inputs.workflowBranch == 'dev' }}
+    uses: shardeum/github-automation/.github/workflows/reusable-node-ci.yml@dev
+    permissions:
+      issues: write
+      pull-requests: write
+      contents: write
+    with:
+      node-version: ${{ vars.NODE_VERSION }}
+      lint-required: ${{ vars.IS_LINT_REQUIRED == 'true' }}
+      format-check-required: ${{ vars.IS_FORMAT_CHECK_REQUIRED == 'true' }}
```
Comment (chrypnotoad) [inline .github/workflows/ci.yml:59]:
```suggestion
      lint-required: ${{ vars.IS_LINT_REQUIRED == 'false' }}
      format-check-required: ${{ vars.IS_FORMAT_CHECK_REQUIRED == 'false' }}
```

File: CODEOWNERS [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+# Please submit requests for new teams to Systems and Automation
+
+# Global approval (all files)
+# * @shardeum/team-name
+
+# Directory-level approval
+/.github/ @shardeum/systems-and-automation
+
+# Specific file rules
+# README.md @shardeum/team-name
```
Comment (chrypnotoad) [inline CODEOWNERS]:
```suggestion
# README.md @shardeum/team-name

```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    uses: shardeum/github-automation/.github/workflows/reusable-node-ci.yml@main
+    permissions:
+      issues: write
+      pull-requests: write
+      contents: write
+    with:
+      node-version: ${{ vars.NODE_VERSION }}
+      lint-required: ${{ vars.IS_LINT_REQUIRED == 'true' }}
+      format-check-required: ${{ vars.IS_FORMAT_CHECK_REQUIRED == 'true' }}
+      apply-patches-required: ${{ vars.IS_APPLY_PATCHES_REQUIRED == 'true' }}
```
Comment (chrypnotoad) [inline .github/workflows/ci.yml:75]:
```suggestion
      format-check-required: ${{ vars.IS_FORMAT_CHECK_REQUIRED == 'false' }}
      apply-patches-required: ${{ vars.IS_APPLY_PATCHES_REQUIRED == 'false' }}
```

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    if: ${{ github.event.inputs.workflowBranch == 'dev' }}
+    uses: shardeum/github-automation/.github/workflows/reusable-node-ci.yml@dev
+    permissions:
+      issues: write
+      pull-requests: write
+      contents: write
+    with:
+      node-version: ${{ vars.NODE_VERSION }}
+      lint-required: ${{ vars.IS_LINT_REQUIRED == 'false' }}
+      format-check-required: ${{ vars.IS_FORMAT_CHECK_REQUIRED == 'false' }}
```
Comment (chrypnotoad) [inline .github/workflows/ci.yml]:
```suggestion
      lint-required: ${{ vars.IS_LINT_REQUIRED == 'true' }}
      format-check-required: ${{ vars.IS_FORMAT_CHECK_REQUIRED == 'true' }}
```

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+    permissions:
+      issues: write
+      pull-requests: write
+      contents: write
+    with:
+      node-version: ${{ vars.NODE_VERSION }}
+      lint-required: ${{ vars.IS_LINT_REQUIRED == 'true' }}
+      format-check-required: ${{ vars.IS_FORMAT_CHECK_REQUIRED == 'false' }}
+      apply-patches-required: ${{ vars.IS_APPLY_PATCHES_REQUIRED == 'false' }}
+      unit-tests-required: ${{ vars.IS_UNIT_TESTS_REQUIRED == 'true' }}
```
Comment (chrypnotoad) [inline .github/workflows/ci.yml]:
```suggestion
      lint-required: ${{ vars.IS_LINT_REQUIRED == 'true' }}
      format-check-required: ${{ vars.IS_FORMAT_CHECK_REQUIRED == 'true' }}
      apply-patches-required: ${{ vars.IS_APPLY_PATCHES_REQUIRED == 'true' }}
      unit-tests-required: ${{ vars.IS_UNIT_TESTS_REQUIRED == 'true' }}
```

**Review by arhamj — DISMISSED** [review]

**Comment by chrypnotoad** [comment]
DO NOT MERGE. Will be merged by Andrew.

**Pull request merged**